test/health_checker_unittest.cc

```diff
@@ -0,0 +1,72 @@
+#include "../src/health_checker.h"
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include <sstream>
+#include <boost/filesystem.hpp>
+
+
+namespace google {
+
+class HealthCheckerUnittest : public ::testing::Test {
+ protected:
+  static void Cleanup(HealthChecker* health_checker){
+    health_checker->CleanupForTest();
+  }
+
+  static void SetUnhealthy(HealthChecker* health_checker,
+                           const std::string& state_name) {
+    health_checker->SetUnhealthy(state_name);
+  }
+
+  static bool IsHealthy(const HealthChecker& health_checker) {
```

Comment:
Why did we add the wrapper here?

Reply:
It's a private function, and C++ `friend` does not extend to subclasses.

Reply:
Ah ok. Thanks.